src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationsCleanupTask.kt

```diff
@@ -18,20 +20,25 @@ internal class RecommendationsCleanupTask(
   @Value("\${mrd.api.url}") private val mrdApiUrl: String? = null,
   private val environment: Environment? = null,
 ) {
-//  @Scheduled(initialDelay = 0, fixedRate = 1814400000) // Initial delay: 0 milliseconds, Fixed rate: 21 days
-//  @Scheduled(cron = "0 30 19 1/21 * ?") // The second schedule will run at 7:30 PM every 21 days starting from the first day of the month
-  @Transactional
+  @Scheduled(initialDelay = 0, fixedRate = 900000) // Fixed rate: 15 minutes (900000 milliseconds), remove this line after the first deployment
+  @Scheduled(cron = "0 0 * 1/1 * ?") // The second schedule will run every hour after the initial 15-minute interval
```

Comment:
Remember we have multiple instances of these things running in prod.  These jobs might end up running on each server at the same time.

Reply:
We now have one schedule. And unfortunately, on the locking using SELECT FOR UPDATE/ @TransaMultictional(isolation = SERIALIZABLE) will only apply to one pod as Robin mentions :( It will not protect against other pods running the schedule at the same time.

* Robin suggested using a distributed locking mechanism -  https://manib.hashnode.dev/distributed-locking-mechanism
* another cheaper/quicker solution would be to lower the number of records selected for deletion by using a LIMIT 2, and perhaps re-introduce the slowDelete(..) function (Darren saw this earlier)